File: src/main/java/com/wikia/tibia/repositories/EffectRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.EffectGateway;
import com.wikia.tibia.objects.Effect;

public class EffectRepository extends WikiObjectRepository {

    public EffectRepository() {
        super(Effect.class, new EffectGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/EffectRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Effect

class EffectRepository : WikiObjectRepository<Effect>(
  wikiObjectClass = Effect::class.java,
  wikiObjectGateway = WikiObjectGateway(Contract.EFFECTS)
)
